kdtree/include/pcl/kdtree/impl/kdtree_flann.hpp

```diff
@@ -120,39 +120,43 @@ pcl::KdTreeFLANN<PointT, Dist>::setInputCloud (const PointCloudConstPtr &cloud,
     return;
   }
 
-  flann_index_.reset (new FLANNIndex (::flann::Matrix<float> (cloud_.get (), 
-                                                              index_mapping_.size (), 
+  flann_index_.reset (new FLANNIndex (::flann::Matrix<float> (cloud_.get (),
+                                                              index_mapping_.size (),
                                                               dim_),
                                       ::flann::KDTreeSingleIndexParams (15))); // max 15 points/leaf
   flann_index_->buildIndex ();
 }
 
 ///////////////////////////////////////////////////////////////////////////////////////////
-template <typename PointT, typename Dist> int 
-pcl::KdTreeFLANN<PointT, Dist>::nearestKSearch (const PointT &point, int k, 
-                                                std::vector<int> &k_indices, 
+template <typename PointT, typename Dist> int
+pcl::KdTreeFLANN<PointT, Dist>::nearestKSearch (const PointT &point, int k,
+                                                std::vector<int> &k_indices,
                                                 std::vector<float> &k_distances) const
 {
   assert (point_representation_->isValid (point) && "Invalid (NaN, Inf) point coordinates given to nearestKSearch!");
+  assert ((k >= 0) && "Invalid number of neighbors");
```

Comment:
Why not just accept unsigned int as input? See #4350 (only the changes in octree module)

Reply:
Good call. Done.